dora/src/compiler/asm.rs

```diff
@@ -263,15 +265,27 @@ where
     }
 
     pub fn int_shl(&mut self, mode: MachineMode, dest: Reg, lhs: Reg, rhs: Reg) {
-        self.masm.int_shl(mode, dest, lhs, rhs);
+        if has_x_ops() {
```

Comment:
This should probably move to `masm/x64.rs` since this is x64-specific. For example ARM64 doesn't have different shift operations AFAIK.

Reply:
Ah yes, will do.